features/step_definitions/branch_steps.rb

```diff
@@ -80,8 +80,20 @@
 end
 
 
+Given(/^I am on the "(.+?)" branch with "(.+?)" as the previous Git branch/) do |current_branch, previous_branch|
+  step %(I am on the "#{previous_branch}" branch)
+  step %(I am on the "#{current_branch}" branch)
+end
+
+
 
 
+Then(/^my previous Git branch is (?:now|still) "(.+?)"/) do |previous_branch|
+  run 'git checkout -'
+  expect(current_branch_name).to eql previous_branch
+  run 'git checkout -'
```

Comment:
How about just reading the previous branch instead of performing a couple checkouts similar to how its done in the source